File: findJudge.js
/**
 * @param {number} N
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function (N, trust) {
  let counts = Array(N).fill(0);
  for (let [truster, trusted] of trust) {
    counts[truster - 1]--;
    counts[trusted - 1]++;
  }
  let judge = -1;
  for (let i = 0; i < counts.length; i++) {
    if (counts[i] === N - 1) {
      if (judge !== -1) return -1;
      judge = i + 1;
    }
  }
  return judge;
};

File: leetcode/findJudge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        counter = [0] * n
        for truster, trusted in trust:
            counter[truster - 1] -= 1
            counter[trusted - 1] += 1

        for i, count in enumerate(counter):
            if count == n - 1:
                return i + 1
        return -1
